relational_operators/AggregationOperator.hpp

```diff
@@ -103,23 +109,23 @@ class AggregationWorkOrder : public WorkOrder {
    * @brief Constructor
    *
    * @param input_block_id The block id.
-   * @param aggr_state_index The index of the AggregationState in QueryContext.
+   * @param state The AggregationState to use.
    **/
   AggregationWorkOrder(
       const block_id input_block_id,
-      const QueryContext::aggregation_state_id aggr_state_index)
+      AggregationOperationState *state)
```

Comment:
Not exact.

For the single node version (no remote `Worker`s at all), `RelationalOperator::getAllWorkOrders` constructs the actual `WO`s with all the info they need. No `WO` serialization involves.

The distributed version (w/ some remote `Worker`s) that requires `WO` serialization has the following new components:
 * `WorkOrderProtosContainer` for normal `WO` protos only,
 * a new `RelationalOperator::getAllNormalWorkOrderProtos`,
 * `Shiftboss`
 * `WorkOrderFactory`, and
 * `CatalogCache`, a local cache for the master `Catalog`.

`Foreman` chooses `WO` protos from `WorkOrderProtosContainer`, and sends them to `Shiftboss`, which then constructs `WO`s from their protos with the access to `QueryContext`, `StorageManager`, and `CatalogCache`.